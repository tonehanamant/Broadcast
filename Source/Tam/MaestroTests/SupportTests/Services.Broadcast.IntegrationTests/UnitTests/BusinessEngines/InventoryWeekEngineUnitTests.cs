using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using Services.Broadcast.BusinessEngines;
using System;
using Tam.Maestro.Data.Entities;

namespace Services.Broadcast.IntegrationTests.UnitTests.BusinessEngines
{
    public class InventoryWeekEngineUnitTests
    {
        private InventoryWeekEngine _InventoryWeekEngine;

        [SetUp]
        public void Setup()
        {
            _InventoryWeekEngine = new InventoryWeekEngine();
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void GetDateRangeInventoryIsAvailableForForWeek_MediaWeekSameAsEffectiveAndEndDate()
        {
            var effectiveDate = new DateTime(2020, 03, 01);
            var endDate = new DateTime(2020,03,31);
            var mediaWeek = new MediaWeek { StartDate = new DateTime(2020, 03, 01), EndDate = new DateTime(2020, 03, 31) };

            var result = _InventoryWeekEngine.GetDateRangeInventoryIsAvailableForForWeek(mediaWeek, effectiveDate, endDate);

            Approvals.Verify(IntegrationTestHelper.ConvertToJson(result));
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void GetDateRangeInventoryIsAvailableForForWeek_MediaWeekSmallerThanEffectiveAndEndDate()
        {
            var effectiveDate = new DateTime(2020, 03, 01);
            var endDate = new DateTime(2020, 03, 31);
            var mediaWeek = new MediaWeek { StartDate = new DateTime(2020, 03, 05), EndDate = new DateTime(2020, 03, 25) };

            var result = _InventoryWeekEngine.GetDateRangeInventoryIsAvailableForForWeek(mediaWeek, effectiveDate, endDate);

            Approvals.Verify(IntegrationTestHelper.ConvertToJson(result));
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void GetDateRangeInventoryIsAvailableForForWeek_EffectiveDateBiggerThanMediaWeek()
        {
            var effectiveDate = new DateTime(2020, 03, 5);
            var endDate = new DateTime(2020, 03, 31);
            var mediaWeek = new MediaWeek { StartDate = new DateTime(2020, 03, 01), EndDate = new DateTime(2020, 03, 31) };

            var result = _InventoryWeekEngine.GetDateRangeInventoryIsAvailableForForWeek(mediaWeek, effectiveDate, endDate);

            Approvals.Verify(IntegrationTestHelper.ConvertToJson(result));
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void GetDateRangeInventoryIsAvailableForForWeek_EndDateSmallerThanMediaWeek()
        {
            var effectiveDate = new DateTime(2020, 03, 1);
            var endDate = new DateTime(2020, 03, 25);
            var mediaWeek = new MediaWeek { StartDate = new DateTime(2020, 03, 01), EndDate = new DateTime(2020, 03, 31) };

            var result = _InventoryWeekEngine.GetDateRangeInventoryIsAvailableForForWeek(mediaWeek, effectiveDate, endDate);

            Approvals.Verify(IntegrationTestHelper.ConvertToJson(result));
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void GetDateRangeInventoryIsAvailableForForWeek_EndDateBiggerThanMediaWeek()
        {
            var effectiveDate = new DateTime(2020, 03, 05);
            var endDate = new DateTime(2020, 04, 25);
            var mediaWeek = new MediaWeek { StartDate = new DateTime(2020, 03, 01), EndDate = new DateTime(2020, 03, 31) };

            var result = _InventoryWeekEngine.GetDateRangeInventoryIsAvailableForForWeek(mediaWeek, effectiveDate, endDate);

            Approvals.Verify(IntegrationTestHelper.ConvertToJson(result));
        }
    }
}
