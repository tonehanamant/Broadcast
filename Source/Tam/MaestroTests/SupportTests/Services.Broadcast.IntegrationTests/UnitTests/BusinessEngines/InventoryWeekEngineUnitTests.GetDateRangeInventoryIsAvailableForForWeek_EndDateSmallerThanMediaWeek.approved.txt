{
  "End": "2020-03-25T00:00:00",
  "Start": "2020-03-01T00:00:00"
}